modules/images/webp-uploads/load.php

```diff
@@ -704,3 +704,184 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
+				return webp_uploads_restore_image( $attachment_id, $data );
+		}
+	}
+
+	return $data;
+}
+add_filter( 'wp_update_attachment_metadata', 'webp_uploads_update_attachment_metadata', 10, 2 );
+
+/**
+ * Before saving the metadata of the image store a backup values for the sources and file property
+ * those files would be used and deleted by the backup mechanism, right after the metadata has
+ * been updated. It removes the current sources property due once this function is executed
+ * right after an edit has taken place and the current sources are no longer accurate.
+ *
+ * @since n.e.x.t
+ *
+ * @param int   $attachment_id The ID representing the attachment.
+ * @param array $data          The current metadata of the attachment.
+ * @return array The updated metadata for the attachment.
+ */
+function webp_uploads_backup_sources( $attachment_id, $data ) {
+	$metadata = wp_get_attachment_metadata( $attachment_id );
+	// Nothing to back up.
+	if ( ! isset( $metadata['sources'] ) ) {
+		return $data;
+	}
+
+	$target = isset( $_REQUEST['target'] ) ? sanitize_text_field( $_REQUEST['target'] ) : 'all';
+
+	// When an edit to an image is only applied to a thumbnail there's nothing we need to back up.
+	if ( 'thumbnail' === $target ) {
+		return $data;
+	}
+
+	$sources = $metadata['sources'];
+	// Prevent execution of the callbacks more than once if the callback was already executed.
+	$has_been_processed = false;
+
+	$hook = function ( $meta_id, $post_id, $meta_name ) use ( $attachment_id, $sources, &$has_been_processed ) {
+		// Make sure this hook is only executed in the same context for the provided $attachment_id.
+		if ( $post_id !== $attachment_id ) {
+			return;
+		}
+
+		// This logic should work only if we are looking at the meta key: `_wp_attachment_backup_sizes`.
+		if ( '_wp_attachment_backup_sizes' !== $meta_name ) {
+			return;
+		}
+
+		if ( $has_been_processed ) {
+			return;
+		}
+
+		$has_been_processed = true;
+		webp_uploads_backup_full_image_sources( $post_id, $sources );
+	};
+
+	add_action( 'added_post_meta', $hook, 10, 3 );
+	add_action( 'updated_post_meta', $hook, 10, 3 );
+
+	// Remove the current sources as at this point the current values are no longer accurate.
+	// TODO: Requires to be updated from https://github.com/WordPress/performance/issues/158.
+	unset( $data['sources'] );
+
+	return $data;
+}
+
+/**
+ * Stores the provided sources for the attachment ID in the `_wp_attachment_backup_sources`  with
+ * the next available target if target is `null` no source would be stored.
+ *
+ * @since n.e.x.t
+ *
+ * @param int   $attachment_id The ID of the attachment.
+ * @param array $sources       An array with the full sources to be stored on the next available key.
+ */
+function webp_uploads_backup_full_image_sources( $attachment_id, $sources ) {
+	$target = webp_uploads_get_next_full_size_key_from_backup( $attachment_id );
+
+	if ( null === $target || empty( $sources ) ) {
+		return;
+	}
```

Comment:
If sources are empty, we don't need to call the `webp_uploads_get_next_full_size_key_from_backup` function. Let's move the sources check to be done before calling the function:
```php
	if ( empty( $sources ) ) {
		return;
	}

	$target = webp_uploads_get_next_full_size_key_from_backup( $attachment_id );
	if ( null === $target ) {
		return;
	}
```